pkg/p2p/libp2p/internal/breaker/breaker.go

```diff
@@ -46,22 +45,25 @@ type breaker struct {
 	backoff              time.Duration // initial backoff duration
 	maxBackoff           time.Duration
 	failInterval         time.Duration // consecutive failures are counted if they happen within this interval
+	currentTimeFn        currentTimeFn
 	mtx                  sync.Mutex
 }
 
 type Options struct {
-	Limit        int
-	FailInterval time.Duration
-	StartBackoff time.Duration
-	MaxBackoff   time.Duration
+	Limit         int
+	FailInterval  time.Duration
+	StartBackoff  time.Duration
+	MaxBackoff    time.Duration
+	CurrentTimeFn currentTimeFn
```

Comment:
This is not right. This will expose this to all users of this package, which is not what we want.

This should be handled similar to the blocklist change. You can have a new constructor which initializes this timeFunc in export_test file. This will only make it available to the tests.